crypto/s2n_dhe.c

```diff
@@ -275,13 +277,13 @@ int s2n_dh_compute_shared_secret_as_server(struct s2n_dh_params *server_dh_param
     Yc.data = s2n_stuffer_raw_read(Yc_in, Yc.size);
     notnull_check(Yc.data);
 
-    pub_key = BN_bin2bn(( const unsigned char * )Yc.data, Yc.size, NULL);
+    DEFER_CLEANUP(pub_key = BN_bin2bn(( const unsigned char * )Yc.data, Yc.size, NULL), BN_free);
```

Comment:
The cleanup function would be `BN_free_pointer`
Ref: https://github.com/awslabs/s2n/blob/main/utils/s2n_safety.h#L339

Reply:
Implemented as suggested.